gmso/core/topology.py

```diff
@@ -504,13 +426,13 @@ def update_connection_types(self):
                     self._bond_types_idx[c.connection_type] = len(self._bond_types) - 1
                 if isinstance(c.connection_type, AngleType):
                     self._angle_types[c.connection_type] = c.connection_type
-                    self._angle_types_idx[c.connection_type] = len(self._angle_types) - 1
+                    self._angle_types_idx[c.connection_type] = len(self._bond_types) - 1
```

Comment:
I think you mean `_angle_type` here? (and also those below)

Reply:
My bad. This should be `_bond_types` and everything below should be updated accordingly.